lib/client/local_proxy_middleware.go

```diff
@@ -170,6 +177,13 @@ func (c *CertChecker) GetOrIssueCert(ctx context.Context) (tls.Certificate, erro
 	return c.cert, nil
 }
 
+// RetrieveError retrieves the happened on while retrieving certificates.
+func (c *CertChecker) RetrieveError() error {
```

Comment:
nit: the description is an incomplete sentence. could we also state the fact this is the last error? like `LastIssueCertError`